gnomad_qc/v5/sample_qc/identify_trios.py

```diff
@@ -0,0 +1,575 @@
+"""Script to identify trios from relatedness data and filter based on Mendel errors and de novos."""
+
+import argparse
+import json
+import logging
+from collections import defaultdict
+from typing import Dict, Optional, Tuple
+
+import hail as hl
+from gnomad.sample_qc.relatedness import (
+    create_fake_pedigree,
+    filter_to_trios,
+    get_duplicated_samples,
+    get_duplicated_samples_ht,
+    infer_families,
+)
+from gnomad.utils.vcf import SEXES
+from hail.utils.misc import new_temp_file
+
+from gnomad_qc.resource_utils import check_resource_existence
+from gnomad_qc.v4.sample_qc.identify_trios import families_to_trios
+from gnomad_qc.v5.resources.basics import (
+    WORKSPACE_BUCKET,
+    add_project_prefix_to_sample_collisions,
+    get_aou_vds,
+    get_logging_path,
+)
+from gnomad_qc.v5.resources.meta import meta, sample_id_collisions
+from gnomad_qc.v5.resources.sample_qc import (
+    dense_trios,
+    duplicates,
+    finalized_outlier_filtering,
+    ped_filter_param_json_path,
+    ped_mendel_errors,
+    pedigree,
+    relatedness,
+    sample_rankings,
+    trios,
+)
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("identify_trios")
+logger.setLevel(logging.INFO)
+
+
+def filter_relatedness_ht(ht: hl.Table, filter_ht: hl.Table) -> hl.Table:
+    """
+    Filter relatedness Table to only include pairs of samples that are both AoU genomes and not QC-filtered.
+
+    :param ht: Relatedness Table.
+    :param filter_ht: Outlier filtering Table.
+    :return: Filtered relatedness Table.
+    """
+    ht = ht.filter(
+        ((ht.i.data_type == "genomes") & (ht.i.project == "aou"))
+        & ((ht.j.data_type == "genomes") & (ht.j.project == "aou"))
+    )
+    ht = ht.key_by(i=ht.i.s, j=ht.j.s)
+
+    # Remove all pairs with a QC-filtered sample.
+    ht = ht.filter(
+        filter_ht[ht.i].outlier_filtered | filter_ht[ht.j].outlier_filtered,
+        keep=False,
```

Comment:
Lets filter to high_quality from the sample qc meta instead considering a hard filtered sample snuck through to outlier filtering somehow?